testng-core/src/test/java/test/retryAnalyzer/RetryAnalyzerTest.java

```diff
@@ -240,6 +243,20 @@ public void ensureRetryDoesntRunEndlesslyForDataDrivenTests() {
     assertThat(test.retryAnalyzer.dataprovider.issue2163.RetryAnalyzer.logs).hasSize(21);
   }
 
+  @Test(description = "GITHUB-2669")
+  public void testFailedRetryWithParameters() {
+    XmlSuite suite = createXmlSuite("GITHUB_2669");
```

Comment:
You can add a new helper method `createXmlSuite(String name, Map<String, String> params)`